sdk/storage/azure-storage-datamovement/inc/azure/storage/datamovement/blob_folder.hpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <string>
+
+#include <azure/storage/blobs.hpp>
+
+namespace Azure { namespace Storage { namespace DataMovement {
+
+  class BlobFolder final {
+  public:
+    static BlobFolder CreateFromConnectionString(
+        const std::string& connectionString,
+        const std::string& blobContainerName,
+        const std::string& blobName,
+        const Blobs::BlobClientOptions& options = Blobs::BlobClientOptions());
+
+    explicit BlobFolder(
+        const std::string& blobUrl,
+        std::shared_ptr<StorageSharedKeyCredential> credential,
+        const Blobs::BlobClientOptions& options = Blobs::BlobClientOptions());
+
+    explicit BlobFolder(
+        const std::string& blobUrl,
+        std::shared_ptr<Core::Credentials::TokenCredential> credential,
+        const Blobs::BlobClientOptions& options = Blobs::BlobClientOptions());
+
+    explicit BlobFolder(
+        const std::string& blobUrl,
+        const Blobs::BlobClientOptions& options = Blobs::BlobClientOptions());
+
+    BlobFolder GetBlobFolder(const std::string& folderName);
+
+    Blobs::BlobClient GetBlobClient(const std::string& blobName);
+  };
+
+  BlobFolder GetBlobFolderFromBlobContainer(
+      const Blobs::BlobContainerClient& blobContainerClient,
+      const std::string& folderName);
```

Comment:
We are usually against methods directly created at namespace level.  Can you move this to be inside a FactoryClass (with no constructor and with this method as public static)

Reply:
I don't know. It looks good to me.

Everything in this subdirectory is temporary. It's not final. I'll just use it for now.

Reply:
You should then use a `/feature` branch for this PR.
Pushing to /main repo compromise a little bit the code that is shipped next to other GA packages.
And there's no benefit from having the changes on /main if it is still temporal. You can get CI gates from /feature branches

Reply:
Ok, created a new branch `feature/dmlib`